logstash-core/lib/logstash/pipeline.rb

```diff
@@ -107,16 +107,23 @@ def compile_lir
     LogStash::Compiler.compile_sources(sources_with_metadata, @settings)
   end
 
-  def plugin(plugin_type, name, *args)
+  def plugin(plugin_type, name, line, column, *args)
     @plugin_counter += 1
 
     # Collapse the array of arguments into a single merged hash
     args = args.reduce({}, &:merge)
 
-    id = if args["id"].nil? || args["id"].empty?
-      args["id"] = "#{@config_hash}-#{@plugin_counter}"
-    else
-      args["id"]
+    # Pull the ID from LIR to keep IDs consistent between the two representations
+    id = lir.graph.vertices.filter do |v| 
```

Comment:
I suppose we could, but IMHO generated code is harder to maintain than non-generated code, so I'd rather have it in here. 

I agree with you from an architectural perspective, but from a pragmatic code-maintenance perspective I disagree.

WRT the separation of concerns, the pipeline class already has more problems than I can count, I don't think this patch makes it materially worse, and it will go away in short order once LIR-native execution becomes a thing.